docs/docs/reference/contextual/extension-methods.md

```diff
@@ -20,272 +21,229 @@ circle.circumference
+  extension [T: Numeric](x: T)
+    def + (y: T): T = summon[Numeric[T]].plus(x, y)
 ```
 
-If an extension method has type parameters, they come immediately after the `def` and are followed by the extended parameter. 
-When calling a generic extension method, any explicitly given type arguments follow the method name. So the `second` method can be instantiated as follows:
+If an extension method has type parameters, they come immediately after `extension` and are followed by the extended parameter.
+When calling a generic extension method, any explicitly given type arguments follow the method name. So the `second` method could be instantiated as follows.
 ```scala
-List(1, 2, 3).second[Int]
+  List(1, 2, 3).second[Int]
 ```
-(it's only a showcase, the compiler could of course infer the type).
+Of course, the type argument here would usually be left out since it can be inferred.
 
-### Extension Instances
 
-It is quite common to wrap one or more extension methods in a given instance,
-in order to make them available as methods without needing to be imported explicitly.
-This pattern is supported by a special `extension` syntax. Example:
-```scala
-extension ops {
-  def (xs: Seq[String]).longestStrings: Seq[String] = {
-    val maxLength = xs.map(_.length).max
-    xs.filter(_.length == maxLength)
-  }
-  def (xs: Seq[String]).longestString: String = xs.longestStrings.head
-  def [T](xs: List[T]).second: T = xs.tail.head
-}
-```
-An extension instance can only contain extension methods. Other definitions are not allowed. The name `ops`
-of the extension is optional. It can be left out:
+Extensions can also take using clauses. For instance, the `+` extension above could equivalently be written with a using clause:
 ```scala
-extension {
-  def (xs: Seq[String]).longestStrings: Seq[String] = ...
-  def [T](xs: List[T]).second: T = ...
-}
+  extension [T](x: T)(using n: Numeric[T])
+    def + (y: T): T = n.plus(x, y)
 ```
-If the name of an extension is not explicitly given, it is synthesized from the name and type of the first implemented extension method.
-
-Extension instances map directly to given instances. The `ops` extension above
-would expand to
-```scala
-given ops as AnyRef {
-  def (xs: Seq[String]).longestStrings: Seq[String] = ...
-  def (xs: Seq[String]).longestString: String = ...
-  def [T](xs: List[T]).second: T = ...
-}
-```
-The type "implemented" by this given instance is `AnyRef`, which
-is not a type one can summon by itself. This means that the instance can
-only be used for its extension methods.
 
 ### Collective Extensions
 
 Sometimes, one wants to define several extension methods that share the same
-left-hand parameter type. In this case one can "pull out" the common parameters into the extension instance itself. Examples:
+left-hand parameter type. In this case one can "pull out" the common parameters into
+a single extension and enclose all methods in braces or an indented region following a '`:`'.
```

Comment:
Yes, we do support that.

Reply:
We should mention that the following are equivalent 
```scala
extension (x: Int) {
  def f(y: Int) = x + y
}
```

```scala
extension (x: Int):
  def f(y: Int) = x + y
```

This would help users understand the scope of the `extension`.